### ISSUE TITLE
Revamp `PasswordInput` behavior for accessibility

### ISSUE THREAD

**Issue by mcmxcdev** [mantinedev/mantine#3627]
### What package has an issue

@mantine/core

### Describe the bug

We use the `PasswordInput` component and there seems to be accessibility issues with it.

Why is the password visibility icon using a `tabIndex` of `-1` by default? How should a keyboard user toggle the visibility then?

We have set the `toggleTabIndex={0}` to it, to work around the issue but then we encounter this issue when running with the [axe extension](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd): 

```
Ensure aria-hidden elements are not focusable nor contain focusable elements

Focusable content should be disabled or be removed from the DOM
```

That's because the `visibilityToggleIcon` has `aria-hidden=true` set.

***

So either:
- keep `aria-hidden=true` on visiblityToggle button AND `tabIndex={-1}` (how it's currently, but then no keyboard navigation)
- remove `aria-hidden` from visiblityToggle button so `toggleTabIndex` can be 0 according to a11y specs

Basically, if `toggleTabIndex` prop is set to zero by the user, remove `aria-hidden` attribute. Maybe also change the default for the icon to be reachable for keyboard users.

### What version of @mantine/hooks page do you have in package.json?

5.10.4

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by rtivital** [comment]
This issue is closed for one of these reasons:

- It was fixed in 7.0
- It is no longer reproducible in 7.0
- It is no longer applicable in 7.0
- It does not have a reproduction

If you think that this issue was closed by mistake, and it is still an issue in version 7.0, please reopen it.
